src/setfit/data.py

```diff
@@ -15,69 +15,115 @@
-    elif dataset == "amazon_counterfactual_en":
-        return {
-            "text": ["The sentence is not counterfactual"] * sample_size
-            + ["The sentence is counterfactual"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "SentEval-CR":
-        return {
-            "text": ["The sentence is negative"] * sample_size + ["The sentence is positive"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "sst5":
-        return {
-            "text": ["The sentence is very negative"] * sample_size
-            + ["The sentence is negative"] * sample_size
-            + ["The sentence is neutral"] * sample_size
-            + ["The sentence is positive"] * sample_size
-            + ["The sentence is very positive"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size + [2] * sample_size + [3] * sample_size + [4] * sample_size,
-        }
-    elif dataset == "enron_spam":
-        return {
-            "text": ["The sentence is ham"] * sample_size + ["The sentence is spam"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "tweet_eval_stance_abortion":
-        return {
-            "text": ["The sentence is none"] * sample_size
-            + ["The sentence is against"] * sample_size
-            + ["The sentence is favor"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size + [2] * sample_size,
-        }
-    elif dataset == "ade_corpus_v2_classification":
-        return {
-            "text": ["The sentence is not related"] * sample_size + ["The sentence is related"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
+def get_templated_dataset(
+    dataset: Dataset = None,
+    candidate_labels: List[str] = None,
+    reference_dataset: str = None,
+    template: str = "This sentence is {}",
+    sample_size: int = 2,
+    text_column: str = "text",
+    label_column: str = "label",
+    multi_label: bool = False,
+    label_names_column: str = "label_text",
+) -> Dataset:
+    """Create templated examples for a reference dataset or refernece labels.
+
+    If `candidate_labels` is supplied, use it for generating the templates.
+    Otherwise, use the labels loaded from `reference_dataset`.
+
+    If input Dataset is supplied, add the examples to it, otherwise create a new Dataset.
+    The input Dataset is assumed to have a text column with the name `text_column` and a
+    label column with the name `label_column`, which contains one-hot or multi-hot
+    encoded label sequences.
+
+    Args:
+        dataset (`Dataset`): A Dataset to add templated examples to.
+        candidate_labels (`List[str]`, defaults to None): The list of candidate
+        labels to be fed into the template to construct examples.
+        reference_dataset (`str`, defaults to None): A dataset to take labels
```

Comment:
```suggestion
        dataset (`Dataset`, *optional*): A Dataset to add templated examples to.
        candidate_labels (`List[str]`, *optional*): The list of candidate
        labels to be fed into the template to construct examples.
        reference_dataset (`str`, *optional*): A dataset to take labels
```
I'm not 100% sure on the recommend Google docstring syntax, but I think this is it?